libs/agno/agno/agent/agent.py

```diff
@@ -2604,10 +2608,55 @@ def arun(  # type: ignore
         output_schema: Optional[Type[BaseModel]] = None,
         yield_run_response: Optional[bool] = None,  # To be deprecated: use yield_run_output instead
         yield_run_output: Optional[bool] = None,
+        background: Optional[bool] = None,
+        websocket: Optional[Any] = None,
         debug_mode: Optional[bool] = None,
         **kwargs: Any,
     ) -> Union[RunOutput, AsyncIterator[RunOutputEvent]]:
-        """Async Run the Agent and return the response."""
+        """Async Run the Agent and return the response.
+
+        Args:
+            input: The input to the agent.
+            stream: Whether to stream the response.
+            background: If True, run in background with WebSocket streaming.
+            websocket: WebSocket connection for real-time event streaming (required if background=True and stream=True).
+            ... (other args)
+        """
+        # Handle background WebSocket execution
+        if background:
+            from agno.os.managers import WebSocketHandler
+
+            websocket_handler = WebSocketHandler(websocket=websocket) if websocket else None
+
+            if stream and websocket:
+                # Background + Streaming + WebSocket = Real-time events
+                return self._arun_background_stream(  # type: ignore
+                    input=input,
+                    user_id=user_id,
+                    session_id=session_id,
+                    session_state=session_state,
+                    audio=audio,
+                    images=images,
+                    videos=videos,
+                    files=files,
+                    stream_events=stream_events or stream_intermediate_steps or False,
+                    websocket_handler=websocket_handler,
+                    knowledge_filters=knowledge_filters,
+                    add_history_to_context=add_history_to_context,
+                    add_dependencies_to_context=add_dependencies_to_context,
+                    add_session_state_to_context=add_session_state_to_context,
+                    dependencies=dependencies,
+                    metadata=metadata,
+                    output_schema=output_schema,
+                    debug_mode=debug_mode,
+                    **kwargs,
+                )
+            elif stream and not websocket:
```

Comment:
What if self.stream?

I think maybe move these checks a bit further down. You need the run_id, etc regardless. All of the "resolution" logic below this can be before the split